daemon/main.go

```diff
@@ -753,10 +755,10 @@ func runDaemon() {
 	log.Info("Launching node monitor daemon")
 	go d.nodeMonitor.Run(path.Join(defaults.RuntimePath, defaults.EventsPipe), bpf.GetMapRoot())
 
-	// Launch cilium-health in the same namespace as cilium.
-	log.Info("Launching Cilium health daemon")
-	d.ciliumHealth = &health.CiliumHealth{}
-	go d.ciliumHealth.Run()
+	// Watch cilium-health daemon.
+	log.Info("Watching Cilium health daemon")
+	d.ciliumHealth = &healthwatch.CiliumHealth{}
+	go d.ciliumHealth.Watch()
```

Comment:
I do wonder, now that the cilium-health instance that polls is being invoked as a separate container, whether it's useful to have the cilium daemon keep an eye on it, or if we could just drop this check and the related underlying code.

Reply:
I'm on the fence about it, so it's probably simpler to just leave this in place unless someone voices a strong opinion either way.

One of the things I have some concern about is that someone upgrades Cilium from an earlier version without updating the `DaemonSet`, then the new Cilium doesn't run the `cilium-health` instance, but `cilium-agent` still relies on it, then the health checks for `cilium-agent` fail (is this true?), causing Cilium to get into a bad state. That said, this is clearly outside the scope of our upgrade instructions[0], so maybe it's fair to just tell users "well, if you don't follow the instructions, this is what happens".

[0] http://docs.cilium.io/en/latest/install/upgrade/